app/livechat/server/api/lib/visitors.js

```diff
@@ -62,13 +63,40 @@ export async function findChatHistory({ userId, roomId, visitorId, pagination: {
 	});
 
 	const total = await cursor.count();
-
 	const history = await cursor.toArray();
-
-	return {
-		history,
-		count: history.length,
-		offset,
-		total,
-	};
+	if(text == 'null'){
+		return {
+			history,
+			count: history.length,
+			offset,
+			total,
+			
+		};
+		
+	}else{
+		let resultArray=[];
+		Meteor.runAsUser(userId,()=>{
+			for(var i=0; i<history.length; i++){
```

Comment:
Why not use `map()` interaction instead of `for` ?

Reply:
using map now instead of for